### ISSUE TITLE
Bugs in autocorrection by Lint/UnneededCopDisableDirective Cop

### ISSUE THREAD

**Issue by ashmaroli** [rubocop-hq/rubocop#6430]
## Expected behaviors

- Expected `Lint/UnneededCopDisableDirective` Cop to **not swallow** an empty line on auto-correction.
- Expected `Lint/UnneededCopDisableDirective` Cop to **remove unnecessary "enable comment"** on auto-correction.

## Actual behavior

On auto-correction:
- `Lint/UnneededCopDisableDirective` **swallowed the preceding blank line**
- `Lint/UnneededCopDisableDirective` **did not remove the now redundant** enable-comment

## Steps to reproduce the problem

1. Add test file (`./test.rb`)
```ruby
# Dummy Class
class FooBar
  def foo
    'alpha beta'
  end

  # rubocop:disable Metrics/AbcSize
  def bar
    'lorem ipsum'
  end
  # rubocop:enable Metrics/AbcSize
end
```
2. Check for linting errors (`rubocop test.rb`)
```
Offenses:

test.rb:7:3: W: Lint/UnneededCopDisableDirective: Unnecessary disabling of Metrics/AbcSize.
  # rubocop:disable Metrics/AbcSize
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```
3. Correct detected error (`rubocop test.rb -a`)
```
Offenses:

test.rb:7:3: W: [Corrected] Lint/UnneededCopDisableDirective: Unnecessary disabling of Metrics/AbcSize.
  # rubocop:disable Metrics/AbcSize
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```
4. Examine test file content (`./test.rb`)
```ruby
# Dummy Class
class FooBar
  def foo
    'alpha beta'
  end
  def bar
    'lorem ipsum'
  end
  # rubocop:enable Metrics/AbcSize
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
0.60.0
```
